js/src/components/free-listings/configure-product-listings/checkErrors.test.js

```diff
@@ -0,0 +1,352 @@
+/**
+ * Internal dependencies
+ */
+import checkErrors from './checkErrors';
+
+function toRates( ...tuples ) {
+	return tuples.map( ( [ countryCode, rate ] ) => ( {
+		countryCode,
+		rate,
+	} ) );
+}
+
+function toTimes( ...tuples ) {
+	return tuples.map( ( [ countryCode, time ] ) => ( {
+		countryCode,
+		time,
+	} ) );
+}
+
+describe( 'checkErrors', () => {
+	describe( 'Shipping rates', () => {
+		let flatShipping;
+		let manualShipping;
+
+		beforeEach( () => {
+			flatShipping = { shipping_rate: 'flat' };
+			manualShipping = { shipping_rate: 'manual' };
+		} );
+
+		it( 'When the type of shipping rate is an invalid value, should not pass', () => {
```

Comment:
:nail_care: 
WDYT of adding "or missing" to explicitly match what's tested here.
```suggestion
		it( 'When the type of shipping rate is an invalid value or missing, should not pass', () => {
```

Also, we could describe, what "not pass" means. Like, 
```suggestion
		it( 'When the type of shipping rate is an invalid value or missing, should not pass - returned object should contain `shipping_rate` property with an error message.', () => {
```
We could add that to JSDoc as well.

Reply:
The descriptions of "returned object should contain property" are applicable for all individual checks, so I added a comment note on the top level of the test scope. Adjusted by https://github.com/woocommerce/google-listings-and-ads/pull/871/commits/5101eb382c404e929cb0efaab698b064f8af418e